ci/management/commands/generate_reports.py

```diff
@@ -303,3 +352,68 @@ def handle_health_worker_registrations(
                 'Cadre': operator_details.get('role'),
                 'Number of Registrations': count,
             })
+
+    def handle_enrollments(self, sheet, sbm_client, ids_client, start_date,
+                           end_date):
+
+        sheet.set_header([
+            'Message set',
+            'Roleplayer',
+            'Currently enrolled (snapshot at point y)',
+            'Cumulatively enrolled (in last period x)',
+            'Of the cumulatively enrolled (in last period x), number which '
+            'opted out',
+            'Of the cumulatively enrolled (in last period x), the number which'
+            ' completed messages',
+        ])
+
+        subscriptions = self.get_subscriptions(
+            sbm_client,
+            created_before=end_date.isoformat())
+
+        data = {}
+        for idx, subscription in enumerate(subscriptions):
+            messageset = self.get_messageset(
+                            sbm_client, subscription['messageset'])
+            identity = self.get_identity(ids_client, subscription['identity'])
+
+            messageset_name = messageset['short_name'].split('.')[0]
+
+            receiver_role = 'None'
+            if identity:
+                receiver_role = identity.get('details', {}).get(
+                                    'receiver_role', 'None')
+
+            key = '%s_%s' % (messageset_name, receiver_role)
+
+            if key not in data:
+                data[key] = {
+                    'messageset': messageset_name,
+                    'receiver_role': receiver_role,
+                    'total': 0,
+                    'total_period': 0,
+                    'optouts': 0,
+                    'completed': 0
+                }
```

Comment:
We can use nested `defaultdict`s here to avoid the if statements. eg.
```
from collections import defaultdict
from functools import partial

data = defaultdict(partial(defaultdict, int))
data[messageset_name, receiver_role]['total'] += 1
```